index.js

```diff
@@ -1,33 +1,59 @@
 const fetch = require('node-fetch'); // Automatically excluded in browser bundles
 
-async function api(endpoint, token) {
-	token = token ? `&access_token=${token}` : '';
-	const response = await fetch(`https://api.github.com/repos/${endpoint}${token}`);
+// Matches '/<re/po>/tree/<ref>/<dir>'
+const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
+
+function parseResource(res) {
+	if (typeof res === 'string') {
+		try {
+			const parsedUrl = new URL(res);
+			res = {api: 'https://api.github.com'};
+			[, res.user, res.repository, res.ref, res.directory] = urlParserRegex.exec(parsedUrl.pathname);
```

Comment:
This way you can avoid the `try`/`catch`

```suggestion
			[, res.user, res.repository, res.ref, res.directory] = urlParserRegex.exec(parsedUrl.pathname) || [];
			if (res.directory !== 'string') {
				 throw new Error('Unable to parse GitHub URL'); 
			}
```